File: Dart/Slice/all_elements_before_index_in_slice.dart
import "dart:core";

void main() {
  // Create a new list of strings.
  final List<String> randomSlice = <String>["Bob", "Alice", "Joe"];
  // Get all the elements before the given index.
  print(getAllElementsBeforeIndex(randomSlice, 2));
}

// Get all the elements before a given index in a slice
List<dynamic> getAllElementsBeforeIndex(
  final List<dynamic> slice,
  final int index,
) {
  return slice.sublist(0, index);
}

File: Go/Slice/all-elements-before-index-in-slice.go
package main

import (
	"fmt"
)

func main() {
	// Create a slice
	randomSlice := []string{"a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n", "o", "p", "q", "r", "s", "t", "u", "v", "w", "x", "y", "z"}
	// Get all the elements before certain index.
	fmt.Println(allElementsBeforeIndexInSlice(randomSlice, 5))
}

// Get all the elements before a particular index in a slice
func allElementsBeforeIndexInSlice(slice []string, index int) []string {
	return slice[:index]
}
